File: FFinder/app/src/main/java/com/ffinder/android/enums/FCMMessageType.java
package com.ffinder.android.enums;

import com.ffinder.android.utils.Strings;

/**
 * Created by SiongLeng on 2/9/2016.
 */
public enum  FCMMessageType {

    UpdateLocation, UserLocated, NotifyRememberToAddFriend, FriendsAdded,
    IsAliveMsg, Nothing;

    public static FCMMessageType convertStringToFCMMessageType(String input){
        if(Strings.isEmpty(input)) return Nothing;
        else{
            for (FCMMessageType c : FCMMessageType.values()) {
                if (c.name().equals(input)) {
                    return FCMMessageType.valueOf(input);
                }
            }
        }
        return Nothing;
    }

}

File: FFfinder_mac/FFfinder/FFfinder/FCMMessageType.swift
//
//  FCMMessageType.swift
//  FFfinder
//
//  Created by Siong Leng Ho on 16/10/16.
//  Copyright © 2016 Lightweight Studio Apps. All rights reserved.
//

import Foundation
enum FCMMessageType:String{
    
    case UpdateLocation
    case UserLocated
    case IsAliveMsg
    case FriendsAdded
    case Nothing
    
    
    
    
    public static func convertStringToFCMMessageType(_ input:String) -> FCMMessageType{
        if let result:FCMMessageType = FCMMessageType(rawValue: input){
            return result
        }
        else{
            return .Nothing
        }
    }
  
}
